### PR TITLE
Add instruction for remote debugging with PyCharm

### PR DESCRIPTION
**Pull request by noorula**
Instructions for remote debugging with PyCharm.

### CODE REVIEW

File: source/howto/remote-access/pycharm.rst [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    :description: How to attach a remote debugger from PyCharm to a Valohai execution
+
+.. _remote-ssh-pycharm:
+
+Debug from PyCharm
+##################
+
+.. warning::
+
+    This guide assumes you're already familiar with the fundamentals of Valohai and are working with a project that's connected to a Git-repository, and configured on Valohai.
```
Comment (magdapoppins) [inline source/howto/remote-access/pycharm.rst:12]: Nitpick: I don't think git repository is spelled together like that. 🤔

Comment (noorula) [reply source/howto/remote-access/pycharm.rst:12]: It's not. 😅

File: source/howto/remote-access/pycharm.rst [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+And the corresponding ``valohai.yaml`` file:
+
+.. code-block:: yaml
+
+    - step:
+      name: train
+      image: python:3.9
+      command: 
+        - pip install numpy
```
Comment (magdapoppins) [inline source/howto/remote-access/pycharm.rst:44]: Is this missing the pydevd install?

**Review by DrazenDodik — APPROVED** [review]
LGTM - but I don't have PyCharm Pro, so I can't test 🤷

File: source/howto/remote-access/pycharm.rst [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        - python train.py
+
+
+1. Create a new Debug Configuration in PyCharm
+-----------------------------------------------
+
```
Comment (magdapoppins) [inline source/howto/remote-access/pycharm.rst:50]:
```suggestion
Create a new Python Debug Server in PyCharm. The IDE host name should be ``localhost`` and the port should be the port that you want to have listening for incoming debug connections on your local machine (here 1234).
```

File: source/howto/remote-access/pycharm.rst [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+Create a new Python Debug Server in PyCharm. The IDE host name should be ``localhost`` and the Port the port that you want to have listening on your local machine (here 1234).
```
Comment (magdapoppins) [inline source/howto/remote-access/pycharm.rst:51]:
```suggestion
Create a new Python Debug Server in PyCharm. The IDE host name should be ``localhost`` and the port should be the port that you want to have listening to incoming debug connections on your local machine (here 1234).
```

File: source/howto/remote-access/pycharm.rst (hunk 1 above)
Comment (magdapoppins) [reply source/howto/remote-access/pycharm.rst:12]: (Also the ones below)

File: source/howto/remote-access/pycharm.rst [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+-----------------------------------------------
+
+Since ``pydevd_pycharm`` does not have a listen mode, you will need to establish a reverse SSH tunnel between your local machine and Valohai worker instance. 
+You'll get the IP of the worker machine from the Valohai execution logs.
+
+.. code-block:: bash
+
+    ssh -i <PATH-TO-YOUR-PRIVATE-SSH-KEY> <IP-FROM-VALOHAI> -p 2222 -R 1234:localhost:1234 -t /bin/bash
+
+Next, you will see a prompt saying that the filepath on the worker machine cannot be found in project.
```
Comment (magdapoppins) [inline source/howto/remote-access/pycharm.rst:123]:
```suggestion
You might see a prompt saying that the filepath on the worker machine cannot be found in project. 
```

File: source/howto/remote-access/pycharm.rst [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+This happens because the path mapping was not defined when creating the debugging configuration. You can just click on the "Auto detect" to set the mapping. 
+
+.. image:: /_images/ssh_remote_debug_pycharm_path.png
+    :alt: Choose Auto detect for the path mapping
+
+Finally, you should see a message in the Debugger console saying that the connection was established. 
+
+.. note:: 
+
+    If you created the SSH connection without the reverse portforwarding, you don't need to drop it but you can adjust the parameters from SSH's internal shell:
```
Comment (magdapoppins) [inline source/howto/remote-access/pycharm.rst:133]:
```suggestion
    If you created the SSH connection without reverse port forwarding, you don't need to interrupt the connection in order to adjust the parameters from SSH's internal shell:
```

File: source/howto/remote-access/pycharm.rst [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+            time.sleep(10)
+
+
+    for x in range(1,20) :
+        print(f"Doing computation {x}")
+        data = np.random.random((50,50))
+        sum = np.sum(data)
+        time.sleep(2)
+
+Now update the ``valohai.yaml`` file to install ``pydevd_pycharm`` in your Valohai execution.
```
Comment (noorula) [inline source/howto/remote-access/pycharm.rst:92]:
```suggestion
Now update the ``valohai.yaml`` file to install ``pydevd_pycharm`` in your Valohai execution. You can copy the correct version from the Run Configuration section in PyCharm.
```

File: source/howto/remote-access/pycharm.rst (hunk 1 above)
Comment (noorula) [inline source/howto/remote-access/pycharm.rst:12]:
```suggestion
    This guide assumes you're already familiar with the fundamentals of Valohai and are working with a project that's connected to a Git repository, and configured on Valohai.
```

File: source/howto/remote-access/pycharm.rst [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+    # The script is halted here, until a debugger is attached
+    connected = False
+    while not connected:
+        try:
+            # The arbitrary container port 1234 on the worker machine is set to connect with the debugger.
+            # Note that this is not the same port you specified during step 1 as that is on your local machine. 
+            pydevd_pycharm.settrace('localhost', port=1234, stdoutToServer=True, stderrToServer=True)
+            connected = True
+        except Exception as e:
+            time.sleep(10)
```
Comment (magdapoppins) [inline source/howto/remote-access/pycharm.rst:83]: We could also
- shorten this to something like 3 seconds 
- add a print("waiting for connections")
- yeet that e as it isn't used
